modules/host/storageobligations.go

```diff
@@ -775,6 +782,19 @@ func (h *Host) threadedHandleActionItem(soid types.FileContractID) {
 			}
 			return
 		}
+		// If the obligation has no sector roots, we can remove the obligation and not
+		// submit a storage proof. The host payout for a failed empty contract
+		// includes the contract cost and locked collateral.
+		if len(so.SectorRoots) == 0 {
+			h.log.Debugln("storage proof not submitted for empty contract, id", so.id())
+			h.mu.Lock()
+			err := h.removeStorageObligation(so, obligationSucceeded)
+			h.mu.Unlock()
+			if err != nil {
+				h.log.Println("Error removing storage obligation:", err)
+			}
+			return
+		}
```

Comment:
I think there is a chance that `len(so.SectorRots) == 0` and `so.proofDeadline() < blockHeight` are both `true`. In that case we probably also want to indicate a success for the storage obligation. That means the two `if` blocks should probably be reversed in order.